File: targets/js/sample-counter-v3/lib/mvu/widgets/button.js
export class Button {
    _label;
    _onPressed;
    constructor(label, onPressed) {
        this._label = label;
        this._onPressed = onPressed;
    }
    build() {
        const btn = document.createElement("button");
        btn.innerHTML = this._label;
        btn.onclick = (_) => this._onPressed();
        return btn;
    }
}

File: samples/SampleCounterV3/Mvu/Widgets/Button.cs
using Metano.TypeScript.DOM;

namespace SampleCounterV3.Mvu.Widgets;

public sealed class Button(string label, Action onPressed) : Widget
{
    public override HtmlElement Render()
    {
        var btn = Js.Document.CreateElement(HtmlElementType.Button);
        btn.TextContent = label;
        btn.OnClick = _ => onPressed();

        return btn;
    }
}
